### PR TITLE
singularity task

### PR DESCRIPTION
**Pull request by djarecka**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary

- [x] fixing singularity task and adding first set of tests

- [x] adding tests with input_spec and output_spec

- [x] adding singularity tests to CI

## Checklist

- [x] All tests passing
- [x] I have added tests to cover my changes
- [ ] I have updated documentation (if necessary)
- [x] My code follows the code style of this project 
(we are using `black`: you can `pip install pre-commit`, 
run `pre-commit install` in the `pydra` directory 
and `black` will be run automatically with each commit)

## Acknowledgment
- [x] I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

**Comment by djarecka** [comment]
small comments to the travis changes:

- I failed to install go in the pre-install part, had to use go as the language in travis and install python in the pre-install. I didn't want to do it for all runs, so I only included singularity in one run

- was having problem with singularity testing failures when `pytest -n auto` (seems to me that one test was downloading image, and the other was failing because couldn't find the image), so I removed `-n` for now

File: .travis.yml [hunk 1]
```diff
@@ -36,10 +35,15 @@ matrix:
...
     env: JOBQUEUE="slurm"
-  allow_failures:
+  - language: go
+    go: 1.13
+    env:
+      - JOBQUEUE="singularity"
+      - INSTALL_TYPE="develop"
   - python: 3.7
     env: INSTALL_DEPENDS="pip==10.0.1 setuptools==30.3.0"
-  - python: 3.8
```
Comment (satra) [inline .travis.yml]: any reason to drop this?

Comment (djarecka) [reply .travis.yml]: not really, just wanted removed something. 3.8 is removed wit pip==10.0.1 only, there are still 7 other runs with 3.8

Comment (satra) [reply .travis.yml]: thank you.

**Comment by djarecka** [comment]
@satra - singularity tests work on openmind after cleaning my home dir :)

**Pull request merged**